### ISSUE TITLE
Testing Styles Modularization

### ISSUE THREAD

**Issue by lexa-diky** [kotest/kotest#3588]
TLDR
Modularize testing styles classes to allow restricting which styles could be used for a project

Suggestion
Modularize testing styles to allow adding them to the project independently one by one

Reasoning
Most of the projects need to choose 1 particular testing style and keep it for most or, even more often, for all tests. This approach promotes code style unification and allows for faster onboarding and easier-to-understand code due to smaller space for 'unique' approaches.

Cons: 
harder style discovery
need to manage 10 more modules for each testing style
This change will require major update and will require all users to make changes to their codebase

**Comment by Kantis** [comment]
I don't think it's a good idea to do this in Kotest. It would be a breaking change for every project using Kotest. 

I suggest using static analysis tools to forbid usage of the styles you want to exclude instead. You could do it by configuring Detekt's [ForbiddenImport](https://detekt.dev/docs/rules/style/#forbiddenimport) for instance.

**Comment by lexa-diky** [comment]
Yes, this is definitely breaking change, and something possible to be done in next major version only. But it will only require users to add new dependencies without changing existing code.

Detekt solution is ok, but it does not solve discoverability issues, I noticed that kotest might be very intimidating thing to use for fairly new Kotlin devs.

**Comment by LeoColman** [comment]
We could add a project configuration to allow our disallow spec classes, and crash them when a forbidden spec is run

**Comment by LeoColman** [comment]
Another option is to implement a `OurCompanySpec` that either have your own wording or implements one of the specs @lexa-diky

**Comment by sksamuel** [comment]
Closing this because kotest has been around for so long, this would be a major breaking change. It may have been the best way to design this initially, but alas the ship has sailed.
